### PR TITLE
Format yaml files

### LINKED ISSUE #533: Validate format of YAML files
Our YAML files for our github actions use inconsistent format, and it would be nice to have them aligned. In line with the our other discussions, we should add a step as part of one of the CI build target that validates the format. (https://github.com/valkey-io/valkey/blob/72538622ff8f207af22ed1b890c86885b52d8df3/.github/workflows/ci.yml#L24)

My suggestion is to google yamlfmt, but I don't have a strong opinion.

### PR DESCRIPTION
**Pull request by adetunjii**
Closes #533

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -89,3 +89,15 @@ jobs:
         dnf -y install epel-release gcc make procps-ng which
         make -j SERVER_CFLAGS='-Werror'
 
+  format-yaml:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+    - uses: actions/setup-go@v5
```
Comment (PingXie) [inline .github/workflows/ci.yml]: should we pin to a commit for `setup-go` too? @bjosv, thoughts?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -89,3 +89,15 @@ jobs:
...
+      with:
+        go-version: '1.22.4'
+    - name: Format Yaml files
+      run: |
+        go install github.com/google/yamlfmt/cmd/yamlfmt@latest
+        yamlfmt .
```
Comment (PingXie) [inline .github/workflows/ci.yml]: will this go into `./deps` as well?

Comment (adetunjii) [reply .github/workflows/ci.yml]: Yes it formats all the yaml files present in the repository

**Comment by madolson** [comment]
I'm a little confused because I thought in the daily.yml we had some inconsistently formatted files. The CI flow you added isn't failing though, so it doesn't look like it would catch those inconsistencies.

**Comment by adetunjii** [comment]
> I'm a little confused because I thought in the daily.yml we had some inconsistently formatted files. The CI flow you added isn't failing though, so it doesn't look like it would catch those inconsistencies.



> I'm a little confused because I thought in the daily.yml we had some inconsistently formatted files. The CI flow you added isn't failing though, so it doesn't look like it would catch those inconsistencies.

I might have to format it locally then push. Subsequently, it should fix all inconsistencies

**Comment by madolson** [comment]
> I might have to format it locally then push. Subsequently, it should fix all inconsistencies

Hm, I think I understand better. It looks like it's just formatting it in the CI temporary directory, which happens in a separate directory from the one that will eventually get merged. Ideally we want the step to fail if it encounters any files it needs to format.

**Comment by adetunjii** [comment]
> > I might have to format it locally then push. Subsequently, it should fix all inconsistencies
> 
> Hm, I think I understand better. It looks like it's just formatting it in the CI temporary directory, which happens in a separate directory from the one that will eventually get merged. Ideally we want the step to fail if it encounters any files it needs to format.

Then I think we need a linter instead of a formatter.

**Comment by madolson** [comment]
> Then I think we need a linter instead of a formatter.

Generally a format checker is a subset of linting, since linting is a broad term for code analysis. We should be able to use the format checker you proposed for the linting, and then give instructions for how the end user can use the same formatting tool to fix it.

**Comment by adetunjii** [comment]
> > Then I think we need a linter instead of a formatter.
> 
> Generally a format checker is a subset of linting, since linting is a broad term for code analysis. We should be able to use the format checker you proposed for the linting, and then give instructions for how the end user can use the same formatting tool to fix it.

Okay. I'll have a look.

**Comment by madolson** [comment]
@adetunjii Maybe take a look at https://github.com/valkey-io/valkey/blob/unstable/.github/workflows/clang-format.yml for an example for how we do the style checking with clang.

**Comment by adetunjii** [comment]
@madolson Are we excluding yaml files in the `deps` directory?

**Comment by madolson** [comment]
@adetunjii Yes, exclude the files in deps.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -49,43 +48,79 @@ jobs:
...
+          make -j SERVER_CFLAGS='-Werror'
+
+  format-yaml:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
```
Comment (PingXie) [inline .github/workflows/ci.yml]: let's pin to a commit like we did for other actions

File: .github/workflows/daily.yml [hunk 4]
```diff
@@ -371,106 +371,61 @@ jobs:
...
...alkey-cli -p 8081 REPLICAOF 127.0.0.1 8080 > /dev/null\nwhile [ $(./src/valkey-cli -p 8081 info replication | grep \"master_link_status:down\") ]; do sleep 1; done;\nsleep 1 # wait for the completion of cache reclaim bio\nVMOUT=$(vmtouch -v /tmp/master/dump.rdb)\necho $VMOUT\ngrep -q \" 0%\" <<< $VMOUT \nVMOUT=$(vmtouch -v /tmp/slave/dump.rdb)\necho $VMOUT\ngrep -q \" 0%\" <<< $VMOUT \nCACHE=$(grep -w file /sys/fs/cgroup/memory.stat | awk '{print $2}')\necho \"$CACHE\"\nif [ \"$(( $CACHE-$CACHE0 ))\" -gt \"8000000\" ]; then exit 1; fi\n#magic___^_^___line\necho \"test reboot doesn't increase cache\"\nPID=$(cat /tmp/master/valkey.pid)\nkill -15 $PID\nwhile [ -x /proc/${PID} ]; do sleep 1; done\n./src/valkey-server --daemonize yes --logfile /dev/null --dir /tmp/master --port 8080\nwhile [ $(./src/valkey-cli -p 8080 info persistence | grep \"loading:1\") ]; do sleep 1; done;\nsleep 1 # wait for the completion of cache reclaim bio\nVMOUT=$(vmtouch -v /tmp/master/dump.rdb)\necho $VMOUT\ngrep -q \" 0%\" <<< $VMOUT\nCACHE=$(grep -w file /sys/fs/cgroup/memory.stat | awk '{print $2}')\necho \"$CACHE\"\nif [ \"$(( $CACHE-$CACHE0 ))\" -gt \"8000000\" ]; then exit 1; fi\n#magic___^_^___line\n"
```
Comment (PingXie) [inline .github/workflows/daily.yml]: Is this the result of the yml format? it is a lot harder to read now.

Comment (adetunjii) [reply .github/workflows/daily.yml:399]: Yh. I'll include a line length limit

Comment (adetunjii) [reply .github/workflows/daily.yml]: This is caused by trailing whitespaces on some lines. See: https://github.com/google/yamlfmt/issues/185

**Review by PingXie — APPROVED** [review]
LGTM. Thank you @adetunjii!

**Pull request merged**